test/bin.js

```diff
@@ -28,11 +29,13 @@ describe('bin', function() {
       }
       consoleLog.apply(null, arguments)
     }
+    process.exit = function noop() {}
     process.argv = process.argv.slice(0, 2)
   })
 
   afterEach(function() {
     console.log = consoleLog // eslint-disable-line no-console
+    process.exit = processExit
     // purge yargs cache
     delete require.cache[require.resolve('yargs')]
```

Comment:
Yep, that is exactly what I thought of. I just remembered, that I had an issue with the caching behavior in another case ... Let's forget this. The explicit deletion from `require.cache` seems to work, so let's not overcomplicate this.

_(I have to dig into this deeper some time, there has to be a way to accomplish this with proxyquire's own means... if not, it has to be a bug.)_